### PR TITLE
Bugfix KNOWAGE-8140

### PR DESCRIPTION
**Pull request by BojanSovticEngIT**
Fixed files (all)

### CODE REVIEW

File: knowage-core/src/main/java/it/eng/spagobi/commons/services/KnowageCaptchaServlet.java [hunk 1]
```diff
@@ -64,4 +51,14 @@ private void writeImage(OutputStream outputstream, BufferedImage bufferedimage)
...
+	private int getWidthParameter(HttpServletRequest request) {
+		String widthParam = request.getParameter("width");
+		return widthParam != null ? Integer.parseInt(widthParam) : 200;
+	}
+
+	private int getHeightParameter(HttpServletRequest request) {
+		String heightParam = request.getParameter("height");
+		return heightParam != null ? Integer.parseInt(heightParam) : 50;
+	}
+
```
Comment (davide-zerbetto) [inline knowage-core/src/main/java/it/eng/spagobi/commons/services/KnowageCaptchaServlet.java:63]: This is a change in logic: width and height were retrieved from servlet config, not from request

**Review by davide-zerbetto — CHANGES_REQUESTED** [review]

File: knowage-core/src/main/java/it/eng/spagobi/commons/services/KnowageCaptchaServlet.java (hunk 1 above)
Comment (BojanSovticEngIT) [reply knowage-core/src/main/java/it/eng/spagobi/commons/services/KnowageCaptchaServlet.java]: Fixed

File: knowage/src/main/java/it/eng/spago/dispatching/httpchannel/AdapterHTTP.java [hunk 2]
```diff
@@ -579,19 +577,11 @@ private void setHttpRequestData(HttpServletRequest request, RequestContainer req
...
-	/*
-	 * (non-Javadoc)
-	 *
-	 * @see javax.servlet.GenericServlet#init()
-	 */
-	@Override
-	public void init() throws ServletException {
-		super.init();
-
+	private boolean getSerializedSession() {
```
Comment (kerny3d) [inline knowage/src/main/java/it/eng/spago/dispatching/httpchannel/AdapterHTTP.java:580]: It's ok but pay attention: the init() method is called only once; that means that in the past the value was read only one time at the boot of the application from a source which let you know the value of a changeable configuration. The code was in strong contradiction as you can say: i think nobody knows what is the right way to manage that value. Anyway this is the less worse way to manage this, good job.

File: knowagebirtreportengine/src/main/java/it/eng/spagobi/engines/birt/BirtReportServlet.java [hunk 3]
```diff
@@ -107,7 +107,7 @@
  **/
 public class BirtReportServlet extends HttpServlet {
 
-	private IReportEngine birtReportEngine = null;
+	// private IReportEngine birtReportEngine = null;
```
Comment (kerny3d) [inline knowagebirtreportengine/src/main/java/it/eng/spagobi/engines/birt/BirtReportServlet.java:110]: Feel free to delete commented code.

**Review by kerny3d — COMMENTED** [review]
Need some fixes

**Pull request merged**